backend/danswer/utils/gpu_utils.py

```diff
@@ -0,0 +1,30 @@
+import requests
+from retry import retry
+
+from danswer.utils.logger import setup_logger
+from shared_configs.configs import INDEXING_MODEL_SERVER_HOST
+from shared_configs.configs import INDEXING_MODEL_SERVER_PORT
+from shared_configs.configs import MODEL_SERVER_HOST
+from shared_configs.configs import MODEL_SERVER_PORT
+
+logger = setup_logger()
+
+
+@retry(tries=5, delay=5)
+def gpu_status_request(indexing: bool = True) -> bool:
+    if indexing:
+        model_server_url = f"{INDEXING_MODEL_SERVER_HOST}:{INDEXING_MODEL_SERVER_PORT}"
+    else:
+        model_server_url = f"{MODEL_SERVER_HOST}:{MODEL_SERVER_PORT}"
+
+    if "http" not in model_server_url:
```

Comment:
No need to change but if this is now repeated functionality, we can pull it out into a function